CRM/Contact/BAO/Query.php

```diff
@@ -5164,15 +5164,20 @@ public function dateQueryBuilder(
         $date = "'$date'";
       }
 
+      $this->_tables[$tableName] = $this->_whereTables[$tableName] = 1;
+
+      // handle IS NULL / IS NOT NULL / IS EMPTY / IS NOT EMPTY
+      if ($this->nameNullOrEmptyOp($fieldName, $op, $grouping)) {
```

Comment:
Why not do this in beginning of the function (after line no 5065) since we are anyway going to return if null/empty operators are used.